### PR TITLE
NXP-18118: Create marketplace for MongoDB Extension and configure the…

### PR DESCRIPTION
**Pull request by kevinleturc**
… audit deployment

### CODE REVIEW

File: marketplace/src/main/resources/install.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!-- See documentation at http://doc.nuxeo.com/x/IgIz -->
+<install>
+  @BUNDLES@
+  @LIBS@
+  @PROPERTIES@
+  @TEMPLATES@
+</install>
```
Comment (mincong-h) [inline marketplace/src/main/resources/install.xml]: A newline is missing.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  </repositories>
+
+  <scm>
+    <connection>scm:git:git://github.com/nuxeo/marketplace-mongodb-ext.git</connection>
+    <developerConnection>scm:git:ssh://git@github.com:nuxeo/marketplace-mongodb-ext.git</developerConnection>
+    <url>https://github.com/nuxeo/marketplace-mongodb-ext</url>
+  </scm>
+
+</project>
+
```
Comment (mincong-h) [inline pom.xml]: Please remove the extra line.

File: marketplace/src/main/assemble/assembly.xml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    <nx:rmdups dir="${outdir}/nxr/lib" />
+    <!-- Temporary ZIP Nuxeo NXR to compare with base distribution NXR -->
+    <zip destfile="${outdir}/nxr-${maven.project.version}.zip" basedir="${outdir}/nxr" />
+
+    <!-- Your MP only needs to include files not already provided by the base distribution -->
+    <nx:zipdiff file1="${outdir}/nxr-${maven.project.version}.zip"
+                file2="${outdir}/nuxeo-nxr-server-${nuxeo.distribution.version}.zip"
+                includesfile="${outdir}/includes"
+                excludesfile="${outdir}/excludesfile"
+                patternsetid="mongodb.ext.versus.cap">
```
Comment (troger) [inline marketplace/src/main/assemble/assembly.xml]: `versus.server`

**Review by mincong-h — CHANGES_REQUESTED** [review]
Just very small things to change. I'll let the others to review the configuration part, I'm not yet familiar with it.

File: marketplace/src/main/resources/package.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  <hotreload-support>true</hotreload-support>
+  <require-terms-and-conditions-acceptance>true</require-terms-and-conditions-acceptance>
+  <nuxeo-validation>none</nuxeo-validation>
+  <production-state>testing</production-state>
+  <supported>false</supported>
+  <platforms>
+    <platform>server-@DISTRIB_VERSION@</platform>
+    <platform>server-@DISTRIB_VERSION@-*</platform>
+  </platforms>
+  <optional-dependencies>
```
Comment (troger) [inline marketplace/src/main/resources/package.xml]: Why are those optional dependencies here? Really needed?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  <groupId>org.nuxeo.marketplace.mongodb.ext</groupId>
+  <artifactId>parent</artifactId>
+  <packaging>pom</packaging>
+  <name>Nuxeo MongoDB Extension Marketplace Parent</name>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <modules>
+    <module>marketplace</module>
+  </modules>
```
Comment (troger) [inline pom.xml:17]: What about adding a simple `startup` test to validate that the MP does not break the server after the installation?

**Review by troger — CHANGES_REQUESTED** [review]

**Review by mincong-h — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (kevinleturc) [reply pom.xml:17]: I thought about that, but we need an environment with a MongoDB running somewhere, as we need this addons only for 8.10, maybe to much work to make Jenkins job work in a MultiDB environment as this code will be merged in Nuxeo Platform for 9.10.

Comment (troger) [reply pom.xml:17]: If that's done for 9.1-SNAPSHOT, I'm good :+1:

**Review by troger — APPROVED** [review]

**Pull request merged**